akka-persistence-typed/src/main/scala/akka/persistence/typed/Retention.scala

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.typed
+
+import akka.annotation.InternalApi
+
+/** Retention bridges snapshot and journal behavioral flow.
+  * Can describe the algorithms of retention used by snapshot
+  * and journal logic, where applicable, focusing on that
+  * which is shared.
+  */
+@InternalApi private[persistence] trait Retention
+
+/**
+  * Retention criteria.
+  *
+  * @param snapshotSize        Snapshots are used to reduce playback/recovery times. This defines
```

Comment:
The implementation of this is the idea I want to suggest in a follow up commit - because I don't think message count for retention of messages and snapshot logic is necessarily the best, or the only way we should offer users. Agreed that it is odd to see part of an idea in this PR but it seemed to need some of it now. I could back it out to simplify.

Reply:
I think we need to flesh out that suggestion before we make it influence the API here. I also find `snapshotSize` too abstract and difficult to understand what it means. If we need two or more different values it should probably be different RetentionCritiera subclasses with concrete naming rather than reusing one field for several different things.

Reply:
Yes, this is not meant to be an implementation of the idea here.